idlparser/src/ffi/types/mod.rs

```diff
@@ -0,0 +1,77 @@
+use crate::types;
+use std::{slice, str};
+
+pub mod visitor;
+
+/// # Safety
+///
+/// See the safity documentation of [`slice::from_raw_parts`].
+#[no_mangle]
+pub unsafe extern "C" fn parse_idl(idl_ptr: *const u8, idl_len: u32) -> *mut Program {
+    let idl = unsafe { slice::from_raw_parts(idl_ptr, idl_len.try_into().unwrap()) };
+    let idl = str::from_utf8(idl).unwrap();
+    let program = types::parse_idl(idl).unwrap();
+    let program = Box::new(program);
+    Box::into_raw(program)
+}
+
+/// # Safety
+///
+/// TODO
+#[no_mangle]
+pub unsafe extern "C" fn free_program(program: *mut Program) {
+    if program.is_null() {
+        return;
+    }
+    unsafe {
+        drop(Box::from_raw(program));
+    }
+}
+
+pub type Program = types::Program;
```

Comment:
Why type alias instead of `use crate::types::Program`?

Reply:
Wanted to make it consistent like all FFI types are defined here even though sometimes via aliasing. Is there anything wrong with it?